File: .vite/deps_temp_be540386/frontend/blogs.py
from flask import Flask, render_template
import requests

app = Flask(__name__)

# Hashnode GraphQL API endpoint
HASHNODE_API_URL = "https://api.hashnode.com/"
USERNAME = "your-hashnode-username"

# Function to fetch blog posts from Hashnode
def fetch_hashnode_blogs(username):
    query = """
    query($username: String!) {
      user(username: $username) {
        publication {
          posts {
            title
            brief
            slug
            coverImage
            dateAdded
          }
        }
      }
    }
    """
    variables = {
        "username": username
    }

    # Make a POST request to Hashnode's GraphQL API
    response = requests.post(HASHNODE_API_URL, json={"query": query, "variables": variables})
    
    if response.status_code == 200:
        data = response.json()
        return data['data']['user']['publication']['posts']
    else:
        return None

# Route to display the blog section
@app.route('/blogs')
def blogs():
    posts = fetch_hashnode_blogs(USERNAME)
    if posts:
        return render_template('blogs.html', posts=posts)
    else:
        return "Failed to retrieve posts."

if __name__ == '__main__':
    app.run(debug=True)

File: .vite/deps_temp_be540386/frontend/blogs.js
// BlogSection.js
import React, { useEffect, useState } from 'react';

const BlogSection = () => {
  const [blogs, setBlogs] = useState([]);

  useEffect(() => {
    const fetchBlogs = async () => {
      const response = await fetch('http://localhost:5000/api/blogs');
      const data = await response.json();
      setBlogs(data);
    };

    fetchBlogs();
  }, []);

  return (
    <div>
      <h2>Latest Blogs</h2>
      <div className="blog-list">
        {blogs.map((blog, index) => (
          <div key={index} className="blog-card">
            <img src={blog.coverImage} alt={blog.title} className="blog-image" />
            <h3>{blog.title}</h3>
            <p>{blog.brief}</p>
            <a href={`https://your-hashnode-username.hashnode.dev/${blog.slug}`} target="_blank" rel="noopener noreferrer">
              Read more
            </a>
            <p>{new Date(blog.dateAdded).toDateString()}</p>
          </div>
        ))}
      </div>
    </div>
  );
};

export default BlogSection;
